notebooks/analyze_looping_across_lending_protocols.ipynb

```diff
@@ -271,20 +278,6 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "all_users = set()\n",
-    "\n",
-    "# Extract and split the collateral and debt information for each protocol\n",
-    "for protocol, df in dataframes.items():\n",
-    "    all_users.update(df['User'])  # Add all users to the set of all users\n",
-    "    for token in tokens:\n",
-    "        borrowed_assets.setdefault(token, set()).update(df[df['Debt'].str.contains(token, na=False)]['User'])\n",
-    "        collateral_assets.setdefault(token, set()).update(df[df['Collateral'].str.contains(token, na=False)]['User'])\n",
-    "\n",
-    "# Find users who borrowed an asset on one protocol and deposited the same asset as collateral on another protocol\n",
-    "common_users = {}\n",
-    "for token in tokens:\n",
-    "    common_users[token] = borrowed_assets[token].intersection(collateral_assets[token])\n",
-    "\n",
     "# Count unique users for each token\n",
     "user_counts = {token: len(users) for token, users in common_users.items()}\n",
     "\n",
```

Comment:
Compare and visualize the "looping" users against active users (users who have deposited or borrowed something).

Reply:
Done! ✅